src/mmw/js/src/draw/views.js

```diff
@@ -500,32 +402,74 @@ var AoIUploadView = Marionette.ItemView.extend({
     }
 });
 
-var SelectAreaView = Marionette.ItemView.extend({
-    $label: $('#boundary-label'),
+var DrawToolBaseView = Marionette.ItemView.extend({
+    template: drawToolTmpl,
 
     ui: {
-        items: '[data-tile-url]',
-        button: '#predefined-shape',
-        helptextIcon: 'a.help'
+        drawToolButton: '.draw-pane-list-item-button',
+        helptextIcon: 'a.help',
+        resetButton: '.reset-draw-button'
     },
 
     events: {
-        'click @ui.items': 'onItemClicked'
+        'click @ui.drawToolButton': 'onClickItem',
+        'click @ui.resetButton': 'resetDrawingState'
     },
 
     modelEvents: {
-        'change': 'render',
-        'change:toolsEnabled': 'removeBoundaryLayer'
+        'change': 'render'
+    },
+
+    initialize: function(options) {
+        this.resetDrawingState = options.resetDrawingState;
     },
 
     templateHelpers: function() {
+        var activeDrawTool = this.model.get('activeDrawTool'),
+            activeDrawToolItem = this.model.get('activeDrawToolItem'),
+            currentZoomLevel = App.getLeafletMap().getZoom(),
+            activeTitle = null,
+            activeInfo = null,
+            toolData = this.getToolData();
+
+        if (activeDrawTool === toolData.id) {
+            var activeItem = _.find(toolData.items, function(item) {
+                return item.id === activeDrawToolItem;
+            })
```

Comment:
Consider using this shorthand `_.find(toolData.items, { id: activeDrawToolItem })`